src/gsy_e/models/strategy/load_hours.py

```diff
@@ -512,12 +515,31 @@ def assign_hours_of_per_day(self, hrs_of_day: List[int], hrs_per_day: List[int])
             raise ValueError("Length of list 'hrs_of_day' must be greater equal 'hrs_per_day'")
 
     def _update_energy_requirement_future_markets(self):
+        if not GlobalConfig.FUTURE_MARKET_DURATION_HOURS:
+            return
+        for time_slot in self.area.future_market_time_slots:
+            if self._allowed_operating_hours(time_slot):
+                current_day = self._get_day_of_timestamp(time_slot)
+                desired_energy_Wh = self.energy_per_slot_Wh
+                if (current_day not in self.hrs_per_day or
+                        self.hrs_per_day[current_day] <= FLOATING_POINT_TOLERANCE):
+                    desired_energy_Wh = 0.0
+            else:
+                desired_energy_Wh = 0.0
+            self.state.set_desired_energy(desired_energy_Wh, time_slot, overwrite=True)
+
+    def _update_energy_requirement_in_state(self):
+        self._update_energy_requirement_spot_market()
+        self._update_energy_requirement_future_markets()
+
+    def _update_energy_requirement_spot_market(self):
         self.energy_per_slot_Wh = convert_W_to_Wh(
             self.avg_power_W, self.simulation_config.slot_length)
         desired_energy_Wh = (self.energy_per_slot_Wh
                              if self._allowed_operating_hours(self.area.spot_market.time_slot)
                              else 0.0)
```

Comment:
I have the feeling that we should here substract the energy traded in this slot (in the future)